src/tox/venv.py

```diff
@@ -323,20 +324,27 @@ def finish(self):
     def _needs_reinstall(self, setupdir, action):
         setup_py = setupdir.join("setup.py")
         setup_cfg = setupdir.join("setup.cfg")
-        args = [self.envconfig.envpython, str(setup_py), "--name"]
         env = self._get_os_environ()
-        output = action.popen(
-            args,
-            cwd=setupdir,
-            redirect=False,
-            returnout=True,
-            env=env,
-            capture_err=False,
-        )
-        name = next(
-            (i for i in output.split("\n") if i and not i.startswith("pydev debugger:")),
-            "",
-        )
+
+        if setup_py.exists():
```

Comment:
If the setup.py does not exist, this here https://github.com/tox-dev/tox/blob/954399ba18/src/tox/package/builder/legacy.py#L37 will fail earlier